internal/cmd/server.go

```diff
@@ -0,0 +1,149 @@
+		},
+		&cli.IntFlag{
+			Name:        "port",
+			Usage:       "Choose a port to listen on",
+			Value:       -1,
+			DefaultText: "random",
+		},
+		&cli.DurationFlag{
+			Name:  "inactivity-timeout",
+			Usage: "Automatically shut down after a period without any connected client",
+			Value: time.Minute,
+		},
+		&cli.BoolFlag{
+			Name:  "nats-logging",
+			Usage: "Enable NATS server logging",
+		},
+	},
+	Hidden: true,
+	Action: func(c *cli.Context) error {
+		opts := jobserver.Options{
+			ServerName:        "github.com/datadog/orchestrion server",
+			Port:              c.Int("port"),
+			InactivityTimeout: c.Duration("inactivity-timeout"),
+			EnableLogging:     c.Bool("nats-logging"),
+		}
+		urlFile := c.String("url-file")
+
+		var mu *filelock.Mutex
+		if urlFile != "" {
+			mu = filelock.MutexAt(urlFile)
+			if err := mu.RLock(); err != nil {
+				return cli.Exit(fmt.Errorf("unable to acquire read lock on %q: %w", urlFile, err), 1)
+			}
+
+			if urlData, err := os.ReadFile(urlFile); err == nil && len(urlData) > 0 {
+				url := string(urlData)
+				conn, err := client.Connect(url)
+				if err == nil {
+					defer conn.Close()
+					return cli.Exit(fmt.Sprintf("A job server is already available at %q", url), 2)
+				}
+			}
+
+			// Upgrade to a write lock
+			if err := mu.Lock(); err != nil {
+				return cli.Exit(fmt.Errorf("unable to acquire write lock on %q: %w", urlFile, err), 1)
+			}
+
+			// At this stage, we're actually going to start, and we will clean up the urlFile after ourselces. This includes
+			// cases when the server is killed by an interrupt signal (Control+C). Note the NATS server has its own signal
+			// handler and will attempt to shut itself down gracefully so we are not doing this here...
+			sigChan := make(chan os.Signal, 1)
+			defer close(sigChan)
+			go func() {
+				for range sigChan {
+					os.Remove(urlFile)
+				}
+			}()
+			signal.Notify(sigChan, os.Interrupt)
+			// We also clean up the file on regular exit
+			defer os.Remove(urlFile)
+		}
+
+		// Start the server for real now...
+		server, err := jobserver.New(&opts)
+		if err != nil {
+			return cli.Exit(fmt.Errorf("failed to start NATS server: %w", err), 1)
+		}
+
+		if urlFile != "" {
+			// Now, write the server's URL to the file...
+			if err := os.WriteFile(urlFile, []byte(server.ClientURL()), 0o644); err != nil {
+				return cli.Exit(fmt.Errorf("failed to write URL file at %q: %w", urlFile, err), 1)
+			}
+
+			// Unlock the URL file if it was locked...
+			if mu != nil {
+				if err := mu.Unlock(); err != nil {
+					return cli.Exit(fmt.Errorf("failed to unlock %q: %w", urlFile, err), 1)
+				}
+			}
+
+			// Watch for removal of the URL file, and shut down the server if/when that happens... Note that if we are unable
+			// to watch, we'll leave it entirely up to the inactivity timeout instead...
+			watcher, err := fsnotify.NewWatcher()
+			if err != nil {
+				log.Warnf("Failed to create file watcher: %v\n", err)
```

Comment:
This `if` is very inconspicuous but brings great issues. How about explaining the issue clearly in 2/3 lines for the user?